src/newgrf_spritegroup.h

```diff
@@ -47,6 +47,14 @@ enum SpriteGroupType : uint8_t {
 struct SpriteGroup;
 struct ResolverObject;
 
+/**
+ * Result of resolving sprite groups:
+ * - std::monostate: Failure.
+ * - uint16: Callback result.
+ * - SpriteGroup: ResultSpriteGroup, TileLayoutSpriteGroup, IndustryProductionSpriteGroup
+ */
+using ResolverResult = std::variant<std::monostate, uint16_t, const SpriteGroup *>;
```

Comment:
Would it make sense to do a `using NewGRFCallbackResult = uint16_t`?
Just so it's a bit clearer what it means in the visitors and when doing`std::get_if`.

Reply:
Sounds good. Updated.
I kept it to `newgrf_spritegroup` though. I don't want to propagate that `typedef` to every caller of `GetXxxCallback`.